File: neetcode150/medium/739-Daily-Temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: list) -> list:

        n = len(temperatures)
        res = [0] * n
        stack = [0]

        for i in range(1, n):
            while len(stack) > 0 and temperatures[i] > temperatures[stack[-1]]:
                index = stack.pop()
                res[index] = i - index
            stack.append(i)

        return res

File: neetcode150/medium/739-Daily-Temperatures.cpp
#include <vector>
#include <stack>

using namespace std;

class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temps) {
        int N = temps.size();

        stack<int> st; 
        vector<int> res(N, 0);
        
        for (int i = 0; i < N; i++){
            int curr = temps[i];
            while (!st.empty() && curr > temps[st.top()]) {
                res[st.top()] = i - st.top();
                st.pop();
            }
            st.push(i);
        }
        return res;
    }
};